pandas/core/algorithms.py

```diff
@@ -508,10 +513,17 @@ def factorize(values, sort=False, order=None, na_sentinel=-1, size_hint=None):
         dtype = original.dtype
     else:
         values, dtype, _ = _ensure_data(values)
-        check_nulls = not is_integer_dtype(original)
-        labels, uniques = _factorize_array(values, check_nulls,
+
```

Comment:
use ``pandas.core.dtypes.missing.na_value_for_dtype`` (or maybe add a kwarg to return the underlying value). just want to try to keep this logic in 1 place.

Reply:
Do you know why `na_value_for_dtype(PeriodDtype)` is nan? It should be NaT, right?